mobile/examples/phi-3/ios/LocalLLM/LocalLLM/README.md

```diff
@@ -0,0 +1,97 @@
+# **Local LLM sample application running Phi3-mini on iOS**
+
+## **Steps**
+
+### General prerequisites
+
+See the general prerequisites [here](../../../../../README.md#General-Prerequisites).
+
+For this application, the following prerequisites are preferred:
+
+1. macOS 14+
+
+2. Xcode 15+ (latest Xcode version perferred.)
+
+3. iOS SDK 16.x + (iPhone 14 or iPhone 15 powered by a A16 or A17 preferred)
+
+**Note**: 
+  The current Xcode project contains a built .dylib for ORT and ORT GenAI. The following steps `A, B, C` under `step 1.` for building from source for the libraries are optional.
+  However if you want to build from source to include the latest updates, please use the `step 1.` as a reference.
+
+### 1. Steps to build from source for ONNX Runtime and Generative AI libraries [Optional]
+
+#### **A. Preparation**
+
+1. Install Python 3.10+
+
+2. Install flatbuffers
+  ```
+    pip3 install flatbuffers
+  ```
+
+3. Install [CMake](https://cmake.org/download/)
+
+#### **B. Compiling ONNX Runtime for iOS**
+
+```bash
+
+git clone https://github.com/microsoft/onnxruntime.git
+
+cd onnxruntime
+
+./build.sh --build_shared_lib --ios --skip_tests --parallel --build_dir ./build_ios --ios --apple_sysroot iphoneos --osx_arch arm64 --apple_deploy_target 16.6 --cmake_generator Xcode --config Release
+
+```
+
+***Notice***
+
+  1. Before compiling, you must ensure that Xcode is configured correctly and set it on the terminal
+
+```bash
+
+sudo xcode-select -switch /Applications/Xcode.app/Contents/Developer 
+
+```
+
+  1. ONNX Runtime needs to be compiled based on different platforms. For iOS, you can compile for arm64 or x86_64 based on needs. If you are running an iOS simulator on an Intel mac, compile for x86_64. Use arm64 for an ARM based mac to run the simulator, and to run on an iPhone.
+   
+  2. It is recommended to directly use the latest iOS SDK for compilation. Of course, you can also lower the version to be compatible with past SDKs.
+
+#### **C. Compiling Generative AI with ONNX Runtime for iOS**
+
+```bash
+
+git clone https://github.com/microsoft/onnxruntime-genai
+
+cd onnxruntime-genai
+
+python3 build.py --parallel --build_dir ./build_iphoneos --ios --ios_sysroot iphoneos --ios_arch arm64 --ios_deployment_target 16.6 --cmake_generator Xcode
+
+```
+
+#### **D. Copy over latest header files and required .dylibs built from source**
+
+If you build from source and get the latest .dylibs for ORT and ORT GenAI, please copy the .dylibs over to `mobile\examples\phi-3\ios\LocalLLM\LocalLLM\lib` and copy the latest header files over to `mobile\examples\phi-3\ios\LocalLLM\LocalLLM\header`.
+
+Source header files required including:
+`<ORT_MAIN_SOURCE_REPO>/onnxruntime/core/session/onnxruntime_c_api.h`,
```

Comment:
Can you include the source paths for the two dylib as well?

Reply:
hmm in this optional step, the two dylibs should be built from user? so the source paths depend on their build paths. I mentioned the relative path to copy to for the "two .dylibs".

Reply:
I added a couple lines to explicitly say how the header file structure should look like after copying .dylibs.

Reply:
It's the location of the dylib in the build output that can be confusing. We have 2 explicit build.py commands that we can use as the source of the dylib. 

We can mention that those paths need to be updated based on the actual config and architecture used.